### PR TITLE
fix(game): show mature content warning on mature hub page

### PR DESCRIPTION
**Pull request by wescopeland**
Resolves an issue raised at https://github.com/RetroAchievements/RAWeb/pull/1661#issuecomment-1629765567.

Currently, even though the mature content hub is linked to itself, the mature content warning is not being displayed. This is easily reproducible by visiting the hub using an incognito browser window (id 8323).

With this change, that particular hub will render the warning even when it is not linked to itself.

### CODE REVIEW

File: public/gameInfo.php [hunk 1]
```diff
@@ -87,6 +87,8 @@
             $gate = true;
         }
     }
+} elseif (str_contains($gameTitle, '[Theme - Mature]')) {
```
Comment (Jamiras) [inline public/gameInfo.php:90]: you need to honor the `v` parameter, or a user who hasn't set their profile permission can never access the page.

File: public/gameInfo.php [hunk 2]
```diff
@@ -87,6 +87,8 @@
...
+    $gate = !($userDetails && BitSet($userDetails['websitePrefs'], $matureContentPref));
```
Comment (Jamiras) [inline public/gameInfo.php:91]: Consider improving readability by distributing the not:
```
$gate = !$userDetails || !BitSet(...);
```

Comment (wescopeland) [reply public/gameInfo.php]: Resolved in https://github.com/RetroAchievements/RAWeb/pull/1663/commits/c2190473a6afc0e5fc6ccacd1d7a6f3a524f9b9d.

File: public/gameInfo.php (hunk 1 above)
Comment (wescopeland) [reply public/gameInfo.php]: Thanks, addressed in https://github.com/RetroAchievements/RAWeb/pull/1663/commits/c2190473a6afc0e5fc6ccacd1d7a6f3a524f9b9d.

**Review by Jamiras — APPROVED** [review]

**Pull request merged**